File: LeetCode/2405/main.cpp
class Solution {
public:
  int partitionString(string s) {

    int n = s.length();
    int ans = 0;

    unordered_map<char, bool> mp;

    for (int i = 0; i < n; i++) {
      if (mp.find(s[i]) != mp.end()) {
        ans++;
        mp.clear();
      }
      mp[s[i]] = true;
    }

    return ans + 1;
  }
};

File: LeetCode/2405/main.py
"""
DIFFICULTY : medium
TAGS : string, hash table, greedy
"""


class Solution:
    def partitionString(self, s: str) -> int:
        mp = {}
        ptr = 0
        ans = 1
        while ptr < len(s):
            if s[ptr] not in mp:
                mp[s[ptr]] = True
                ptr += 1
            else:
                mp.clear()
                ans += 1

        return ans
